### DISCUSSION TITLE
Spike Firmware on Mindstorms Hub?

### DISCUSSION THREAD

**Discussion by yipal** [pybricks/support#1158] [Q&A] [answered]
Hi all,
First off, thank you for providing the firmware restoration tool online.  It saved one of my Blue Mindstorms hubs.  

I have a question that is not related to pybricks, but this community might be best equipped to answer it:  

Lego is ending support for the Mindstorms software product, but continues development of the Spike Prime software with v3.  I tried to install the Spike Prime 3 software on my Mindstorms Hub, but the Spike software will not install it, and Lego says they don't support Spike software on Mindstorms Hubs.  

Is it possible to install the "Official Lego Spike Prime Software" on my Blue Mindstorms hub using DFU mode + code.pybricks.com ?
Would that allow me to update my Blue Mindstorms hub with Spike Prime 3?

**Comment by dlech** [comment]
We can only provide support for the Pybricks firmware. We leave it up to the community to hack around and find out what is possible with the official LEGO firmware.

**Comment by laurensvalk** [comment, marked as answer]
While it is technically possible, we don't know if it is allowed by LEGO Education. We prefer to play by the rules to be on the safe side, so we only offer the restore option for version 2.

> First off, thank you for providing the firmware restoration tool online. It saved one of my Blue Mindstorms hubs.

Thanks for letting us know! I'm glad we helped save yet another hub :smile: 

If I may ask, how would you ideally use your LEGO MINDSTORMS hub? I imagine you'd like to upgrade because you'd like to use certain features. Can you share what you are looking for?

We are always working to make Pybricks better. Maybe we can help keep MINDSTORMS alive for everyone :)

**Reply by yipal** [reply]
Thank you, I am involved with two FLL teams, and the teams currently use the Scratch style code tools. I read that the Spike v3 is a bit better and the v3 tool supports bluetooth on ChromeOS Flex (which I wanted to use), so I wanted to try it out v3.

If I install the Spike firmware on the Blue Hub via DFU and v3 still does not work (or something else breaks), would I be able to recover the Blue Hub by reinstalling the Mindstorms firmware?

pybricks does sound intriguing. I have been wanting to switch my older team to python, and the a more responsive python editor sure sounds great; I think I'll try it out.  The younger kids though, I will be keeping on Scratch for a while longer.

**Reply by laurensvalk** [reply]
Right, so what about a Scratch-like interface for Pybricks? It's something we'd like to do in the long run, so we're interested to find out in which ways people would use it.

> would I be able to recover the Blue Hub by reinstalling the Mindstorms firmware?

Yes, the DFU option should always work to recover, no matter what is currently on the hub.

This is why we use DFU in Pybricks, even if it is slightly more cumbersome. But it ensures that it works no matter if people currently used SPIKE or MINDSTORMS.

**Reply by yipal** [reply]
Yes, a Scratch programming model would be helpful. Kids have a much easier time with that style of programming when they are young.
Another thing that makes it easier for them is to have the programming slots on the Hub, like program 0, 1 2, etc.  The kids need to install a different programs for each of the robot runs, and it's easier for them if the infrastructure provides a menu like system for selecting from the different programs.

Also, do you know if there is any difference between the Mindstorms and Spike hardware?

**Reply by dlech** [reply]
> Also, do you know if there is any difference between the Mindstorms and Spike hardware?

The biggest difference is that one is teal and the other is yellow. The bootloader has a different version, so there is some small difference in the bootloader (e.g. it has a different USB product ID). We use the same firmware for both, so there are not any functional differences.

**Reply by yipal** [reply]
Thanks for your help. I installed the Spike firmware on my teal hub from code.pybricks.com and tried to update that to Spike 3 using spike.legoeducation.com but that runs into an error during the update to Spike 3, so I suppose Lego's upgrade tool won't allow Spike 3 on the teal hub through the normal upgrade.
